### PR TITLE
Reduce alglist tree. Fixes #396

### LINKED ISSUE #396: Modify algorithms tab in order to not have three hierarchic levels
Navigating the algorithms tab section is currently a bit cumbersome due to the 3 levels of nesting we are using. Modify it to have just two levels:

- first the algorithm groups
- then each algorithm

Here is a mockup of what it would look like:

geo-ldn-dock-algorithms-tab-two-levels

### PR DESCRIPTION
**Pull request by luipir**

### CODE REVIEW

**Comment by luipir** [comment]
This is an example

Screenshot from 2021-05-20 15-44-55

**Comment by luipir** [comment]
I also removed the fastidious blue ribbon in the tree view

**Comment by luipir** [comment]
and rebased onto decision-tree

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
Does not seem to work very well. Sometimes the group descriptions disappear. Not sure why, but it seems related to the mouse position and whether it is hovering the widget or not.

I recorded a small video that shows this:


https://user-images.githubusercontent.com/732010/119006691-7dab4800-b988-11eb-98bf-b39626e286c0.mp4

**Comment by luipir** [comment]
> Does not seem to work very well. Sometimes the group descriptions disappear. Not sure why, but it seems related to the mouse position and whether it is hovering the widget or not.

Ok... checking

**Review by ricardogsilva — APPROVED** [review]
Looks good now, thanks!

**Comment by luipir** [comment]
@Samweli waiting for your review

**Comment by luipir** [comment]
and someone to merge

File: LDMP/models/algorithms.py
Comment (Samweli) [inline LDMP/models/algorithms.py:132]:
```suggestion
        return 0
```

Comment (Samweli) [reply LDMP/models/algorithms.py:132]: this method should just `return 0`, I feel the `if self.details is None:` is redundant

**Review by Samweli — CHANGES_REQUESTED** [review]
Nice addition to remove the hierarchy. I would like for the algorithm UI remain the same but just removing the trees, I think the tabling of the algorithm items and bolding of the algorithm title complicates the UI feel.

**Comment by luipir** [comment]
> I think the tabling of the algorithm items and bolding of the algorithm title complicates the UI feel.

I almost agree... but adding more text under alg name is confusing. I't not clear if description belogs to the alg or the next one... I'll try to find a cleaner solultion (I completly not a UX guy ;) )

File: LDMP/models/algorithms.py
Comment (luipir) [reply LDMP/models/algorithms.py:132]: I leaved dinstintion of the return just as trace in case want to go back adding subtree with more details.
I would leave the code in thsi way to explict rememeber to do something in case more datails are available . In this case no add any leaf.
If it's important I can simplify, but IMHO is useful for future maintainer of the code.

Comment (luipir) [reply LDMP/models/algorithms.py:132]: I resolve conversation, feel free to open and I'll apply the changes if you disagree with my consideartions

**Comment by luipir** [comment]
@Samweli how is it now?
Screenshot from 2021-05-21 15-13-56

**Comment by Samweli** [comment]
> @Samweli how is it now?

@luipir it looks better now

**Comment by ricardogsilva** [comment]
@luipir @Samweli I was just catching up on these latest changes. I'm not a huge fan of this new frameless option. In my opinion, it now looks a bit messier.

In order for this to work I think we need to differentiate better where one algorithm ends and the other begins. I suggest:

- making the algorithm name and the algorithm description have a bigger difference in font size
- adding a bit more space between the algortihm description and the next algorithm's name

**Comment by luipir** [comment]
> * making the algorithm name and the algorithm description have a bigger difference in font size
> * adding a bit more space between the algortihm description and the next algorithm's name

addessed changes and rebased to decision-tree

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
I think it looks better now. To be honest I'm still not a big fan of this frameless version. But I'm willing to accept it, it can be further refined later, if needed.

The only problem I am gettingis this error, which is now appearing whenever I open the plugin dock:

```python
2021-05-28T15:39:59     WARNING    Traceback (most recent call last):
              File "/home/ricardo/.local/share/QGIS/QGIS3/profiles/trends-earth-dev/python/plugins/LDMP/models/datasets_delegate.py", line 94, in paint
              editorWidget = self.createEditor(self.parent, option, index)
              File "/home/ricardo/.local/share/QGIS/QGIS3/profiles/trends-earth-dev/python/plugins/LDMP/models/datasets_delegate.py", line 121, in createEditor
              return DatasetEditorWidget(item, plugin=self.plugin, parent=parent)
              File "/home/ricardo/.local/share/QGIS/QGIS3/profiles/trends-earth-dev/python/plugins/LDMP/models/datasets_delegate.py", line 171, in __init__
              self.progressBar.setValue(self.dataset.progress)
             TypeError: setValue(self, int): argument 1 has unexpected type 'NoneType'
```

The code still seems to work despite the error though.

**Comment by luipir** [comment]
> The only problem I am gettingis this error, which is now appearing whenever I open the plugin dock:

you have a descriptr that for some reason do not have self.dataset.progress
did you edit it manually for some reason?

**Comment by ricardogsilva** [comment]
No I didn't,. But I did delete my local Trends.Earth directory at some point, as I stated in [another comment](https://github.com/ConservationInternational/trends.earth/pull/430#issuecomment-849697683)

Maybe it is related to that?

**Comment by luipir** [comment]
> No I didn't. But I did delete my local Trends.Earth directory at some point, as I stated in [another comment](https://github.com/ConservationInternational/trends.earth/pull/430#issuecomment-849697683)
> 
> Maybe it is related to that?

better to check what is the datasets that is "malformed" (?) and investigate what workflow generated this issue

**Comment by luipir** [comment]
> The only problem I am gettingis this error, which is now appearing whenever I open the plugin dock:

@ricardogsilva remember to save base_data_directory to try to replicate the issue. Now rebasing  the PR to the latest decision-tree and we can check if the error is still there.

**Comment by luipir** [comment]
> No I didn't. But I did delete my local Trends.Earth directory at some point, as I stated in [another comment](https://github.com/ConservationInternational/trends.earth/pull/430#issuecomment-849697683)
> 
> Maybe it is related to that?

I'm able to replicate.. I'll check it after rebase

**Comment by luipir** [comment]
after rebase disappeared the issue... plase check

**Comment by luipir** [comment]
rebased onto decinsion-tree... ready to merge if the issue discovered by @ricardogsilva has disappeared

**Comment by Samweli** [comment]
> @ricardogsilva @Samweli rebased onto decinsion-tree... ready to merge if the issue discovered by @ricardogsilva has disappeared

@luipir I have tested the latest changes here, I can't seem to reproduce the error.

**Review by ricardogsilva — APPROVED** [review]
Seems to be working fine now. Great work.

LGTM!

**Review by Samweli — APPROVED** [review]

**Pull request merged**
